src/systems/link_plot3d_plugin/LinkPlot3DPlugin.cc

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <ignition/msgs/model.pb.h>
+#include <ignition/plugin/Register.hh>
+#include <ignition/transport.hh>
+#include <ignition/math.hh>
+#include <ignition/msgs.hh>
+
+#include "LinkPlot3DPlugin.hh"
+
+using namespace ignition;
+using namespace gazebo;
+using namespace systems;
+
+//////////////////////////////////////////////////
+LinkPlot3DPlugin::LinkPlot3DPlugin()
+    : System()
+{
+}
+
+//////////////////////////////////////////////////
+void LinkPlot3DPlugin::Configure(const Entity &_entity,
+    const std::shared_ptr<const sdf::Element> &_sdf,
+    EntityComponentManager &_ecm)
+{
+  this->dataPtr->model = Model(_entity);
+
+  if (!this->dataPtr->model.Valid(_ecm))
+  {
+    ignerr << "LinkPlot3DPlugin plugin should be attached to a model "
+           << "entity. Failed to initialize." << std::endl;
+    return;
+  }
+
+  // Get params from SDF
+  this->dataPtr->jointName = _sdf->Get<std::string>("joint_name");
```

Comment:
I recommend getting `link_name`s instead. The joints by themselves are not very helpful to find points in the 3D space.

You could also support multiple links and expose other options like an offset from the link's origin and the material color. See what it looked like on [Gazebo classic](https://github.com/osrf/gazebo/blob/10863ffa8ec28e8e0e3d3a3a588d21cdc78055ab/worlds/plot3d.world#L23-L41). Here, I recommend something like this:

```
      <plugin name='3dplot' filename='LinkPlot3DPlugin'>

        <!-- Update at 10 Hz -->
        <frequency>10</frequency>

        <!-- Upper link plot -->
        <plot>
          <link>double_pendulum_with_base::upper_link</link>

          <!-- Point 1m in +Z axis in link frame -->
          <offset>0 0 1</offset>

          <color>1 0 0</color>
        </plot>

        <!-- Upper link plot -->
        <plot>
          <link>double_pendulum_with_base::lower_link</link>

          <!-- Point 1m in +Z axis in link frame -->
          <offset>0 0 1</offset>

          <color>0 0 1</color>
        </plot>

      </plugin>
```

* Using an `offset` vector instead of `pose` because we don't need rotation
* Using `color` instead of `material` because Ignition doesn't support Ogre materials

Let me know if you have any questions!